--- gstreamer-sharp-0.9.2/configure.ac.new	2011-10-05 19:59:12.017506467 +0200
+++ gstreamer-sharp-0.9.2/configure.ac	2011-10-05 19:59:43.998974689 +0200
@@ -24,7 +24,7 @@ PKG_CHECK_MODULES(MONO_DEPENDENCY, mono
 AM_PROG_LIBTOOL
 
 if test "x$has_mono" = "xtrue"; then
-	GACUTIL_FLAGS='/package $(PACKAGE_VERSION) /gacdir $(libdir) /root $(DESTDIR)/$(libdir)'
+	GACUTIL_FLAGS='/package $(PACKAGE_VERSION) /gacdir $(prefix)/lib /root $(DESTDIR)/$(prefix)/lib'
 	GENERATED_SOURCES=generated/*.cs
 	AC_PATH_PROG(MONO, mono, no)
 	AC_PATH_PROG(CSC, gmcs, no)
